config/config.go

```diff
@@ -34,18 +34,17 @@ import (
 	"github.com/go-kit/log/level"
 	"github.com/miekg/dns"
 	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
 	"github.com/prometheus/common/config"
 )
 
 var (
-	configReloadSuccess = promauto.NewGauge(prometheus.GaugeOpts{
+	configReloadSuccess = prometheus.NewGauge(prometheus.GaugeOpts{
```

Comment:
Rather than switch away from promauto, we should move these vars inside the `SafeConfig` struct. Then we can create them in `NewSafeConfig()`.

Reply:
I moved to `NewSafeConfig` but we can't use `promauto` as it always uses the default registerer (`config_test.go` panics due to duplicated registration)

Reply:
You can use promauto with an explicit registry.